src/index.js

```diff
@@ -38,24 +38,37 @@ const getJobMiddleware = (jobAssertion, jobOperation, errorCode = 400) => async
   await next();
 };
 
-router.get('/api/job', async (ctx, next) => {
+const listJobs = async (ctx, next) => {
   ctx.body = await jobsReady.then(jobs => jobs.toArray());
   await next();
-});
-
-router.post('/api/job', getJobMiddleware(jobAssertions.notExists, jobOperations.create));
-
-router.del('/api/job/:jobName', getJobMiddleware(jobAssertions.alreadyExists, jobOperations.delete));
-
-router.put('/api/job/:jobName', getJobMiddleware(jobAssertions.alreadyExists, jobOperations.update));
-
-router.post('/api/job/once', getJobMiddleware(jobAssertions.alreadyExists, jobOperations.once));
-
-router.post('/api/job/every', getJobMiddleware(jobAssertions.alreadyExists, jobOperations.every));
+};
+const createJob = getJobMiddleware(jobAssertions.notExists, jobOperations.create);
+const removeJob = getJobMiddleware(jobAssertions.alreadyExists, jobOperations.delete);
+const updateJob = getJobMiddleware(jobAssertions.alreadyExists, jobOperations.update);
+const runJobOnce = getJobMiddleware(jobAssertions.alreadyExists, jobOperations.once);
+const runJobEvery = getJobMiddleware(jobAssertions.alreadyExists, jobOperations.every);
+const runJobNow = getJobMiddleware(jobAssertions.alreadyExists, jobOperations.now);
+const cancelJobs = getJobMiddleware(jobAssertions.doNotAssert, jobOperations.cancel);
 
-router.post('/api/job/now', getJobMiddleware(jobAssertions.alreadyExists, jobOperations.now));
+// V1
+router.get('/api/v1/job', listJobs);
+router.post('/api/v1/job', createJob);
+router.del('/api/v1/job/:jobName', removeJob);
+router.put('/api/v1/job/:jobName', updateJob);
+router.post('/api/v1/job/once', runJobOnce);
+router.post('/api/v1/job/every', runJobEvery);
+router.post('/api/v1/job/now', runJobNow);
+router.post('/api/v1/job/cancel', cancelJobs);
 
-router.post('/api/job/cancel', getJobMiddleware(jobAssertions.doNotAssert, jobOperations.cancel));
+// Latest
```

Comment:
Why not have the latest do a 302?

Reply:
That's right, thanks, didn't read the last part of the SO answer. I tested 302 with `Postman` and redirection changed request type to `GET`, seems we should use 307 instead according to [Wikipedia](https://en.wikipedia.org/wiki/HTTP_302):

> Many web browsers implemented this code in a manner that violated this standard, changing the request type of the new request to GET, regardless of the type employed in the original request (e.g. POST).[1] For this reason, HTTP/1.1 (RFC 2616) added the new status codes 303 and 307 to disambiguate between the two behaviours, with 303 mandating the change of request type to GET, and 307 preserving the request type as originally sent. Despite the greater clarity provided by this disambiguation, the 302 code is still employed in web frameworks to preserve compatibility with browsers that do not implement the HTTP/1.1 specification.[2]

Also what I understand from the [SO answer](https://stackoverflow.com/a/398564/1568394) is that we should do it the other way around, i.e. versioned URL should redirect to latest, that correct?